md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/TableColumnLineageAPI.java

```diff
@@ -0,0 +1,155 @@
+            //insert args as received from the jsp
+            String[] args = {"-col", batchId};
+            GetDotForTable getDotForTable = new GetDotForTable();
+
+            String dot = getDotForTable.dotGeneratorWithCol(args);;
+
+            LineageInfo lineageInfo = new LineageInfo();
+            lineageInfo.setDot(dot);
+            lineageInfo.setBatchId(batchId);
+            restWrapper = new RestWrapper(lineageInfo, RestWrapper.OK);
+            LOGGER.info("Record with ID:" + batchId + " selected(getbyBatch) from Lineage by User:" + principal.getName());
+        } catch (Exception e) {
+            restWrapper = new RestWrapper(e.getMessage(), RestWrapper.ERROR);
+        }
+        return restWrapper;
+    }*/
+
+    /**
+     * This method is used to see the relationship between processes linked to a particular instanceExecId.
+     * It also generates the dot string for visualisation of the links.
+     *
+     * @param tableName
+     * @return restWrapper It contains an instance of LineageInfo.
+     */
+    @RequestMapping(value = {"", "/"}, method = RequestMethod.GET)
+    public
+    @ResponseBody
+    RestWrapper getByTableCol(@RequestParam(value = "tableName", defaultValue = "null") String tableName, @RequestParam(value = "colName", defaultValue = "null") String colName, Principal principal) {
+        RestWrapper restWrapper = null;
+        try {
+
+            if(colName != null && !(colName.trim().equals(""))) {
+                LOGGER.info("ColName given: " + colName + "TableName given: " + tableName);
+                String[] args = {colName, tableName};
+                String dot = new String();
+                GetDotForTable getDotForTable = new GetDotForTable();
+                dot = getDotForTable.dotGeneratorWithCol(args);
+                LineageTabColInfo lineageInfo = new LineageTabColInfo();
+                LOGGER.debug(dot);
+                lineageInfo.setDot(dot.toString());
+                lineageInfo.setTableName(tableName);
+                lineageInfo.setColName(colName);
+//            lineageInfo.setInstanceExecId(instanceExecId);
+                restWrapper = new RestWrapper(lineageInfo, RestWrapper.OK);
+                LOGGER.info("Getting " + tableName + "for column name: " + colName + " Lineage by User:" + principal.getName());
+            } else {
+                LOGGER.info("TableName given: " + tableName);
+                String[] args = {tableName};
+                String dot = new String();
+                GetDotForTable getDotForTable = new GetDotForTable();
+                dot = getDotForTable.dotGeneratorWithTable(args);
+                LineageTabColInfo lineageInfo = new LineageTabColInfo();
+                LOGGER.debug(dot);
+                lineageInfo.setDot(dot.toString());
+                lineageInfo.setTableName(tableName);
+
+//            lineageInfo.setInstanceExecId(instanceExecId);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)